app/db.js

```diff
@@ -7,42 +7,48 @@ const nconf = require('../config');
 const dbUrl = nconf.get('db');
 mongoose.Promise = Promise;
 
+let tearingDown = false;
+
 const connect = function () {
-  /**
-   const options = {
-    server: {
-      socketOptions: {keepAlive: 1},
-      auto_reconnect: true
-    }
-  }; */
   const options = {
     useMongoClient: true,
-    // logger: logger,
-    loggerLevel: 'warning' // @todo fetch from config file
+    logger: logger.log.bind(logger),
+    loggerLevel: 'debug' // @todo fetch from config file
   };
-  logger.info(`Create MongoDB connection: ${dbUrl}`);
+
+  // Resolve after database connection is ready and listening
+  const connectionPending = new Promise((resolve) => {
+    mongoose.connection.once('connected', resolve);
+  });
+
+  logger.info(`Connecting to MongoDB: ${dbUrl}`);
   return mongoose
     .connect(dbUrl, options)
-    .then(() => {
-      mongoose.connection.on('error', () => {
-        logger.error(`Could not connect to MongoDB: ${dbUrl}`);
-      });
-    });
+    .then(() => connectionPending);
```

Comment:
is this needed at all? doesn't  mongoose connect resolves when connected..?

Reply:
Looking at the documentation you are quite correct. It is not needed.